activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -803,7 +803,12 @@ def checkout_new_connection
 
         def checkout_and_verify(c)
           c._run_checkout_callbacks do
-            c.verify!
+            ActiveSupport::Notifications.instrumenter.instrument(
```

Comment:
Can we use the existing checkout callbacks ☝🏻 instead of adding this instrumentation?